stockflux-securities-master/src/components/SecuritiesTable.js

```diff
@@ -1,68 +1,69 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import Components from "stockflux-components";
 import "./SecuritiesTable.css";
 
-// hard coded data to be replaced with data from backend
-const SECURITIES_DATA = [
-  { exch: "LSE", symbol: "AAPL", name: "Apple inc." },
-  { exch: "LSE", symbol: "TSLA", name: "Tesla inc." },
-  { exch: "NYSE", symbol: "AAPL", name: "Apple inc." },
-  { exch: "STFX", symbol: "SCOTT", name: "Scott Logic inc." },
-  { exch: "LSE", symbol: "AAPL", name: "Apple inc." },
-  { exch: "LSE", symbol: "TSLA", name: "Tesla inc." },
-  { exch: "NYSE", symbol: "AAPL", name: "Apple inc." },
-  { exch: "STFX", symbol: "SCOTT", name: "Scott Logic inc." },
-  { exch: "LSE", symbol: "AAPL", name: "Apple inc." },
-  { exch: "LSE", symbol: "TSLA", name: "Tesla inc." },
-  { exch: "NYSE", symbol: "AAPL", name: "Apple inc." },
-  { exch: "STFX", symbol: "SCOTT", name: "Scott Logic inc." },
-  { exch: "LSE", symbol: "AAPL", name: "Apple inc." },
-  { exch: "LSE", symbol: "TSLA", name: "Tesla inc." },
-  { exch: "NYSE", symbol: "AAPL", name: "Apple inc." },
-  { exch: "STFX", symbol: "SCOTT", name: "Scott Logic inc." },
-  { exch: "LSE", symbol: "AAPL", name: "Apple inc." },
-  { exch: "LSE", symbol: "TSLA", name: "Tesla inc." },
-  { exch: "NYSE", symbol: "AAPL", name: "Apple inc." },
-  { exch: "STFX", symbol: "SCOTT", name: "Scott Logic inc." },
-  { exch: "LSE", symbol: "AAPL", name: "Apple inc." },
-  { exch: "LSE", symbol: "TSLA", name: "Tesla inc." },
-  { exch: "NYSE", symbol: "AAPL", name: "Apple inc." },
-  { exch: "STFX", symbol: "SCOTT", name: "Scott Logic inc." },
-  { exch: "LSE", symbol: "AAPL", name: "Apple inc." },
-  { exch: "LSE", symbol: "TSLA", name: "Tesla inc." },
-  { exch: "NYSE", symbol: "AAPL", name: "Apple inc." },
-  { exch: "STFX", symbol: "SCOTT", name: "Scott Logic inc." },
-  { exch: "LSE", symbol: "AAPL", name: "Apple inc." },
-  { exch: "LSE", symbol: "TSLA", name: "Tesla inc." },
-  { exch: "NYSE", symbol: "AAPL", name: "Apple inc." },
-  { exch: "STFX", symbol: "SCOTT", name: "Scott Logic inc." },
-  { exch: "STFX", symbol: "SCOTT", name: "Scott Logic inc." },
-  { exch: "LSE", symbol: "AAPL", name: "Apple inc." },
-  { exch: "LSE", symbol: "TSLA", name: "Tesla inc." },
-  { exch: "NYSE", symbol: "AAPL", name: "Apple inc." },
-  { exch: "STFX", symbol: "SCOTT", name: "Scott Logic inc." }
-];
+async function getSecuritiesData() {
+  const response = await fetch("http://ws00128:3001/securities");
+  const securities = await response.json();
+  return securities;
+}
 
 const SecuritiesTable = () => {
+  const [securitiesData, setSecuritiesData] = useState([]);
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getSecuritiesData().then(securities => {
+      setIsLoading(false);
+      setSecuritiesData(securities);
+    });
+    getSecuritiesData().catch(() => {
```

Comment:
You should chain the `.catch` here e.g. `getSecuritiesData().then(...).catch(...)`. The way you have it now, `getSecuritiesData()` will be called twice (2 promises), one with a `then` handler and one with a `catch` handler.

Reply:
Okay, I've made the change now. Could you please check that you think I've done this correctly?

Reply:
Yeah that looks good to me, thanks!